Tests/DatadogTests/DatadogObjc/DDConfigurationTests.swift

```diff
@@ -1,36 +0,0 @@
-/*
-* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
-* This product includes software developed at Datadog (https://www.datadoghq.com/).
-* Copyright 2019-2020 Datadog, Inc.
-*/
-
-import XCTest
-@testable import Datadog
-@testable import DatadogObjc
-
-/// This tests verify that objc-compatible `DatadogObjc` wrapper properly interacts with`Datadog` public API (swift).
-class DDConfigurationTests: XCTestCase {
-    // MARK: - Customizing logs endpoint
-
-    private let swiftEndpoints: [Datadog.Configuration.LogsEndpoint] = [
-        .eu, .us, .custom(url: "https://api.example.com/v1/logs")
-    ]
-    private let objcEndpoints: [DDLogsEndpoint] = [
-        .eu(), .us(), .custom(url: "https://api.example.com/v1/logs")
-    ]
-
-    func testItForwardsLogsEndpointToSwift() {
```

Comment:
This was testing if `DDConfiguration` initialization leads to expected `Datadog.Configuration`. With this removal we lost coverage on `DDLogsEndpoint` → `Datadog.Configuration.LogsEndpoint` mapping:

<img width="853" alt="Screenshot 2020-04-08 at 17 05 17" src="https://user-images.githubusercontent.com/2358722/78800012-3992f900-79bb-11ea-80d4-e59f48bb1f13.png">

Reply:
when i `DDDatadog.initialize` to implicitly create `UploadURLProvider`, i don't see a way to access to it.

the path seems like the following
`Datadog->instance->logsUploadStrategy->uploadWorker->(private) dataUploader->(private) urlProvider`

do we have a better idea than making them `internal` just for tests?

Reply:
`DDConfigurationTests` was only asserting if given `DDConfiguration` builds expected Swift `Datadog.Configuration`, so it doesn't configure nor call `Datadog.initialize` under the hood.

As in this PR we change `Datadog.Configuration` to be basic `struct` with `clientToken` and `logsUploadEndpoint` being the only properties, this should be enough to have 100% coverage on `DDConfiguration`:

```swift
import XCTest
@testable import Datadog
@testable import DatadogObjc

/// This tests verify that objc-compatible `DatadogObjc` wrapper properly interacts with`Datadog` public API (swift).
class DDConfigurationTests: XCTestCase {
    func testItFowardsInitializationToSwift() {
        let objcBuilder = DDConfiguration.builder(clientToken: "abc-123")

        let swiftConfigurationDefault = objcBuilder.build().sdkConfiguration
        XCTAssertEqual(swiftConfigurationDefault.clientToken, "abc-123")
        XCTAssertEqual(swiftConfigurationDefault.logsEndpoint, .us)

        objcBuilder.set(endpoint: .eu())
        let swiftConfigurationEU = objcBuilder.build().sdkConfiguration
        XCTAssertEqual(swiftConfigurationEU.logsEndpoint, .eu)

        objcBuilder.set(endpoint: .us())
        let swiftConfigurationUS = objcBuilder.build().sdkConfiguration
        XCTAssertEqual(swiftConfigurationUS.logsEndpoint, .us)

        objcBuilder.set(endpoint: .custom(url: "https://api.example.com/v1/logs"))
        let swiftConfigurationCustom = objcBuilder.build().sdkConfiguration
        XCTAssertEqual(swiftConfigurationCustom.logsEndpoint, .custom(url: "https://api.example.com/v1/logs"))
    }
}

extension Datadog.Configuration.LogsEndpoint: Equatable {
    public static func == (_ lhs: Datadog.Configuration.LogsEndpoint, _ rhs: Datadog.Configuration.LogsEndpoint) -> Bool {
        switch (lhs, rhs) {
        case (.us, .us): return true
        case (.eu, .eu): return true
        case let (.custom(lhsURL), .custom(rhsURL)): return lhsURL == rhsURL
        default: return false
        }
    }
}
```

💡 Please note that I hand-crafted `Equatable` conformance in this test (vs marking it conform to `Equatable` in `Datadog.framework`) as we rather don't want the `LogsEndpoint` to be `Equatable` in public interface as that would require backward compatibility consideration with each change.

Reply:
copied&pasted ✅ 
but let's reconsider this builder for `v2.0` please ♻️